harness/determined/common/storage/shared.py

```diff
@@ -60,9 +68,32 @@ def restore_path(self, storage_id: str) -> Iterator[str]:
             f"Storage directory does not exist: {self._base_path}. Please verify that you are "
             "using the correct configuration value for checkpoint_storage.host_path",
         )
-        storage_dir = os.path.join(self._base_path, storage_id)
+        storage_dir = os.path.join(self._base_path, src)
+        if not os.path.exists(storage_dir):
+            raise errors.CheckpointNotFound(f"Did not find checkpoint {src} in shared_fs storage")
+        yield pathlib.Path(storage_dir)
+
+    def delete(self, tgt: str) -> None:
+        """
+        Delete the stored data from persistent storage.
+        """
+        storage_dir = os.path.join(self._base_path, tgt)
+
         if not os.path.exists(storage_dir):
-            raise errors.CheckpointNotFound(
-                f"Did not find checkpoint {storage_id} in shared_fs storage"
-            )
-        yield storage_dir
+            raise errors.CheckpointNotFound(f"Storage directory does not exist: {storage_dir}")
+        if not os.path.isdir(storage_dir):
+            raise errors.CheckpointNotFound(f"Storage path is not a directory: {storage_dir}")
+        shutil.rmtree(storage_dir, ignore_errors=False)
+
+    def upload(self, src: Union[str, os.PathLike], dst: str) -> None:
+        src = os.fspath(src)
+        shutil.copytree(src, os.path.join(self._base_path, dst))
+
+    def download(self, src: str, dst: Union[str, os.PathLike]) -> None:
+        dst = os.fspath(dst)
+        try:
+            shutil.copytree(os.path.join(self._base_path, src), dst)
+            return None
+        except FileNotFoundError:
+            pass
+        raise errors.CheckpointNotFound(f"Did not find checkpoint {src} in shared_fs storage")
```

Comment:
```suggestion
        try:
            shutil.copytree(os.path.join(self._base_path, src), dst)
        except FileNotFoundError:
            raise errors.CheckpointNotFound(f"Did not find checkpoint {src} in shared_fs storage")
```

Reply:
I see the clarity of the error messages produced by a library as part of the measure of the quality of a library, and so I dislike giving longer stacktraces than necessary, which is why I wrote it the way I did.

The requests library is guilty of excessively long stacktraces.  They'll give you a 3-step traceback covering a page and a half of your terminal, all to say "yo that request timed out".

So the way I wrote this, we would discard some stacktrace going into shutil.copytree (which is useless) and raise a CheckpointNotFound error with the src in the error message (which is like 99.9% likely to be enough of a stack trace).

Reply:
as per your out-of-band suggestion, I'll change this to `raise CheckpointNotFound() from None` to discard the earlier stacktrace.